dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/DefaultConjureRuntime.java

```diff
@@ -29,6 +31,9 @@
  * {@link DefaultConjureRuntime} provides functionality required by generated handlers.
  */
 public final class DefaultConjureRuntime implements ConjureRuntime {
+    @VisibleForTesting
+    static final CaffeineSpec DEFAULT_SERDE_CACHE_SPEC =
+            CaffeineSpec.parse("maximumSize=1000,expireAfterAccess=1m,softValues");
```

Comment:
I've been burned a few too many times by soft references, my preference is either hard or weak references. Soft references tend to cause gc pressure issues for some collectors, and behave identically to weak references in others (Shenandoah iirc, although it's possible this has been updated).

In this case the value will be held as a strong reference in any existing client that's still used, so I'd opt toward weak values.

Reply:
switched to weak values

Reply:
Thanks!